packages/dds/tree/src/feature-libraries/treeCursorUtils.ts

```diff
@@ -177,7 +177,7 @@ class StackCursor<TNode> extends SynchronousCursor implements CursorWithNode<TNo
 	public enterNode(index: number): void {
 		// assert(this.mode === CursorLocationType.Fields, "must be in fields mode");
 		const siblings = this.getField();
-		assert(index in siblings, 0x405 /* child must exist at index */);
+		assert(index in siblings, "a child does not exist at the specified index, check the status of a node using `Tree.status()`");
```

Comment:
Please throw a `UsageError` instead. Users of SharedTree are not supposed to encounter asserts.
Also, we tend to capitalize the first word of `UsageError` messages.

We may want to change it back to an assert later if we can ensure that users should never end up with a reference to a deleted node, but it seems unlikely we'll manage that for all scenarios, so a `UsageError` is best until we can be sure.